include/seqan3/alignment/pairwise/detail/policy_alignment_result_builder.hpp

```diff
@@ -0,0 +1,111 @@
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    policy_alignment_result_builder() = default; //!< Defaulted.
+    policy_alignment_result_builder(policy_alignment_result_builder const &) = default; //!< Defaulted.
+    policy_alignment_result_builder(policy_alignment_result_builder &&) = default; //!< Defaulted.
+    policy_alignment_result_builder & operator=(policy_alignment_result_builder const &) = default; //!< Defaulted.
+    policy_alignment_result_builder & operator=(policy_alignment_result_builder &&) = default; //!< Defaulted.
+    ~policy_alignment_result_builder() = default; //!< Defaulted.
+
+    /*!\brief Construction and initialisation using the alignment configuration.
+     * \param[in] config The alignment configuration [not used in this context].
+     */
+    policy_alignment_result_builder(alignment_configuration_t const & SEQAN3_DOXYGEN_ONLY(config))
+    {}
+    //!\}
+
+    /*!\brief Builds the seqan3::alignment_result based on the given alignment result type and then invokes the
+     *        given callable with the result.
+     *
+     * \tparam indexed_sequence_pair_t The type of the indexed sequence pair.
+     * \tparam score_t The type of the score.
+     * \tparam callback_t The type of the callback to invoke.
+     *
+     * \param[in] indexed_sequence_pair The indexed sequence pair.
+     * \param[in] score The best alignment score.
+     * \param[in] callback The callback to invoke with the generated result.
+     *
+     * \details
+     *
+     * Generates a seqan3::alignment_result object with the results computed during the alignment. Depending on the
+     * seqan3::align_cfg::result configuration only the requested values are stored. In some cases some additional
+     * work is done to generate the requested result. For example computing the associated alignment from the traceback
+     * matrix.
+     */
+    template <typename indexed_sequence_pair_t, typename score_t, typename callback_t>
+    //!\cond
+        requires std::invocable<callback_t, result_type>
+    //!\endcond
+    void make_result_and_invoke([[maybe_unused]] indexed_sequence_pair_t && indexed_sequence_pair,
+                                [[maybe_unused]] score_t score,
+                                callback_t && callback)
+    {
+        using std::get;
+        using invalid_t = std::nullopt_t *;
+
+        result_type result{};
+
+        static_assert(!std::same_as<decltype(result.data.id), invalid_t>,
+                      "Invalid configuration. Expected result with id!");
+        result.data.id = std::move(get<1>(indexed_sequence_pair));
+
+        if constexpr (traits_type::compute_score)
+        {
+            static_assert(!std::same_as<decltype(result.data.score), invalid_t>,
+                          "Invalid configuration. Expected result with score!");
+            result.data.score = std::move(score);
+        }
+
```

Comment:
```suggestion

       // TODO: Add other result.data like sequence1_begin_position, sequence1_end_position, and so on.

```

Reply:
I don't think this is necessary in this PR. I mean I have the change in front of my eyes. This is nothing I forget until monday.

Reply:
This is better understandable for the reviewer. :shrug: